contracts/BytesLib.sol

```diff
@@ -0,0 +1,509 @@
+// SPDX-License-Identifier: Unlicense
```

Comment:
we shouldn't use unlicensed solutions, consider the comments about `abi.encode` above.